### PR TITLE
Debug "upload-populator-pvc-test not in phase Bound" failure

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

We have seen several PRs affected by this error. This is a test PR to replicate the failures and have more detailed debugging. 

Failures: https://search.ci.kubevirt.io/?search=upload-populator-pvc-test+not+in+phase+Bound+within&maxAge=336h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note

```

### CODE REVIEW

**Comment by alromeros** [comment]
Running the most affected lanes:
/test pull-containerized-data-importer-e2e-ceph
/test pull-containerized-data-importer-e2e-ceph-wffc
/test pull-containerized-data-importer-e2e-hpp-previous
/test pull-containerized-data-importer-e2e-hpp-latest

**Comment by akalenyu** [comment]
So you need a bit more to make the lanes only run 1 test
https://github.com/kubevirt/containerized-data-importer/pull/2384

**Comment by alromeros** [comment]
These two fail too
/test pull-containerized-data-importer-e2e-nfs
/test pull-containerized-data-importer-e2e-istio

**Comment by alromeros** [comment]
The test failed in [hpp latest](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-hpp-latest/1722292735181852672) but the results are odd. The upload pod succeeded and the PVC' is lost, meaning that the PV was rebound. For some odd reason, the target PVC is still pending despite all its fields looking fine.

The claimRef of the PV looks as expected after the rebind:

```yaml
             "claimRef": {
                    "namespace": "cdi-e2e-tests-upload-func-test-tqs2l",
                    "name": "upload-populator-pvc-test",
                    "uid": "e7ca40ed-3237-4c33-8a5e-5b7fc13c1b6d",
                    "resourceVersion": "4249"
                },
```

The only field that doesn't correspond to the one in the target PVC is resourceVersion, but I'm unsure that's enough to prevent binding. @akalenyu do you have any idea on what could be happening?

**Comment by akalenyu** [comment]
> The test failed in [hpp latest](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-hpp-latest/1722292735181852672) but the results are odd. The upload pod succeeded and the PVC' is lost, meaning that the PV was rebound. For some odd reason, the target PVC is still pending despite all its fields looking fine.
> 
> The claimRef of the PV looks as expected after the rebind:
> 
> ```yaml
>              "claimRef": {
>                     "namespace": "cdi-e2e-tests-upload-func-test-tqs2l",
>                     "name": "upload-populator-pvc-test",
>                     "uid": "e7ca40ed-3237-4c33-8a5e-5b7fc13c1b6d",
>                     "resourceVersion": "4249"
>                 },
> ```
> 
> The only field that doesn't correspond to the one in the target PVC is resourceVersion, but I'm unsure that's enough to prevent binding. @akalenyu do you have any idea on what could be happening?

I don't see this log line in the cdi-deployment pod `Rebinding PV to target PVC` which is weird, no?

**Comment by alromeros** [comment]
> > The test failed in [hpp latest](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-hpp-latest/1722292735181852672) but the results are odd. The upload pod succeeded and the PVC' is lost, meaning that the PV was rebound. For some odd reason, the target PVC is still pending despite all its fields looking fine.
> > The claimRef of the PV looks as expected after the rebind:
> > ```yaml
> >              "claimRef": {
> >                     "namespace": "cdi-e2e-tests-upload-func-test-tqs2l",
> >                     "name": "upload-populator-pvc-test",
> >                     "uid": "e7ca40ed-3237-4c33-8a5e-5b7fc13c1b6d",
> >                     "resourceVersion": "4249"
> >                 },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The only field that doesn't correspond to the one in the target PVC is resourceVersion, but I'm unsure that's enough to prevent binding. @akalenyu do you have any idea on what could be happening?
> 
> I don't see this log line in the cdi-deployment pod `Rebinding PV to target PVC` which is weird, no?

Right, maybe we are using a lower log verbosity? I'll add some print statements there.

**Comment by akalenyu** [comment]
> > > The test failed in [hpp latest](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-hpp-latest/1722292735181852672) but the results are odd. The upload pod succeeded and the PVC' is lost, meaning that the PV was rebound. For some odd reason, the target PVC is still pending despite all its fields looking fine.
> > > The claimRef of the PV looks as expected after the rebind:
> > > ```yaml
> > >              "claimRef": {
> > >                     "namespace": "cdi-e2e-tests-upload-func-test-tqs2l",
> > >                     "name": "upload-populator-pvc-test",
> > >                     "uid": "e7ca40ed-3237-4c33-8a5e-5b7fc13c1b6d",
> > >                     "resourceVersion": "4249"
> > >                 },
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The only field that doesn't correspond to the one in the target PVC is resourceVersion, but I'm unsure that's enough to prevent binding. @akalenyu do you have any idea on what could be happening?
> > 
> > 
> > I don't see this log line in the cdi-deployment pod `Rebinding PV to target PVC` which is weird, no?
> 
> Right, maybe we are using a lower log verbosity? I'll add some print statements there.

Nope, we're doing 3 which should make it show up
https://github.com/kubevirt/containerized-data-importer/blob/9464a5417e6aeafc9abb2ec988512e7b19964c48/cluster-sync/sync.sh#L28-L29

**Comment by alromeros** [comment]
I keep investigating this. No major updates other than we can confirm that for some reason there's no rebinding in the upload populator during the failed test. All the logs point to https://github.com/kubevirt/containerized-data-importer/pull/2972/files#diff-ff57e5fda04cf7f099e57af2429b98c52d3334facb36560ef4f4507f6d90959aR1744, so there's something attempting to rebind the target PVC but with a bad resource version. I'll post any updates.

File: tests/upload_test.go [hunk 1]
```diff
@@ -415,38 +415,29 @@ var _ = Describe("[rfe_id:138][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
 				if blockMode {
 					if !f.IsBlockVolumeStorageClassAvailable() {
 						Skip("Storage Class for block volume is not available")
 					}
 					pvcDef = utils.UploadPopulationBlockPVCDefinition(f.BlockSCName)
 				}
+				// Let's see if the PVC exist before creating it
+				pvc, err = utils.FindPVC(f.K8sClient, f.Namespace.Name, pvcDef.Name)
+				if pvc != nil {
+					By("PVC already exists. This isn't good")
```
Comment (akalenyu) [inline tests/upload_test.go:437]: if you look at the Get() impl. https://github.com/kubevirt/containerized-data-importer/blob/00844de42be3b5f65a5a6795b5cda9e9f0954cc9/vendor/k8s.io/client-go/kubernetes/typed/core/v1/persistentvolumeclaim.go#L74 the pvc result is never actually nil

**Comment by akalenyu** [comment]
> I keep investigating [this](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-ceph/1722712895478829056). No major updates other than we can confirm that for some reason there's no rebinding in the upload populator during the failed test. All the logs point to https://github.com/kubevirt/containerized-data-importer/pull/2972/files#diff-ff57e5fda04cf7f099e57af2429b98c52d3334facb36560ef4f4507f6d90959aR1733, so there's something rebinding the target PVC other than the populator. I'll post any updates.
> 
> EDIT: Just noticed that the first [log](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/2972/pull-containerized-data-importer-e2e-ceph/1722712895478829056/artifacts/pods/1_cdi-z54wr_cdi-deployment-96b44f545-8ctvj-cdi-controller.log) line is
> 
> ```
> 2023-11-09T20:47:19Z	DEBUG	events	Successfully uploaded into upload-populator-pvc-test	{"type": "Normal", "object": {"kind":"PersistentVolumeClaim","namespace":"cdi-e2e-tests-upload-func-test-kffz6","name":"upload-populator-pvc-test","uid":"514b75e4-8e1c-450c-9a7c-c2c1e1bded6a","apiVersion":"v1","resourceVersion":"7244"}, "reason": "UploadSucceeded"}
> ```
> 
> So we may be getting the rebinding log, but for some reason the test's output is trimmed. Any reason for this?

One weird thing I noticed is that the test starts at `20:44` according to ginkgo logs
but the first cdi log line we have saved for debugging is
```
2023-11-09T20:47:19Z	DEBUG	events	Successfully uploaded into upload-populator-pvc-test	{"type": "Normal", "object": {"kind":"PersistentVolumeClaim","namespace":"cdi-e2e-tests-upload-func-test-kffz6","name":"upload-populator-pvc-test","uid":"514b75e4-8e1c-450c-9a7c-c2c1e1bded6a","apiVersion":"v1","resourceVersion":"7244"}, "reason": "UploadSucceeded"}
```

**Comment by alromeros** [comment]
Seems that https://github.com/kubevirt/containerized-data-importer/pull/2985 points in the right direction, many thanks @awels and @akalenyu for the great work. Closing this one.

**Pull request closed without merging**